test/ibm/test_ibm_job_states.py

```diff
@@ -398,9 +403,12 @@ def test_transpiling_status(self):
 
     def run_with_api(self, api):
         """Creates a new ``IBMJob`` running with the provided API object."""
-        backend = IBMBackend(mock.Mock(), mock.Mock(), mock.Mock(), api_client=api)
+        simulator_backend = self.provider.get_backend('ibmq_qasm_simulator')
+        backend = IBMBackend(simulator_backend.configuration(),
+                             mock.Mock(), mock.Mock(), api_client=api)
```

Comment:
If all you need from `provider` is a backend configuration, you can just use one of the fake backends from `qiskit-terra` (e.g. `FakeBogota`). Initializing a provider is slow until we have lazy init.